api/v1beta1/gcpmachine_types.go

```diff
@@ -122,6 +122,28 @@ type GCPShieldedInstanceConfig struct {
 	IntegrityMonitoring IntegrityMonitoringPolicy `json:"integrityMonitoring,omitempty"`
 }
 
+// ConfidentialComputePolicy represents the confidential compute configuration for the GCP machine.
+type ConfidentialComputePolicy string
```

Comment:
Do you think we should have this a `bool` in our API instead of exposing `Enabled` and `Disabled`?

Do you think its likely that GCP will add another potential value here in the future?

Reply:
I can't say, fields tend to start as a Boolean value, e.g. ConfidentialComputeEnabled: true|false, and later evolve into needing more states during the API's lifetime.
Enum allows adding a new value if needed.

Reply:
In that case, let's leave it as is.